src/transfer_zip.py

```diff
@@ -0,0 +1,657 @@
+    refs = source_butler.query_datasets(
+        "raw",
+        exposure=exp.id,
+        instrument=instrument,
+        collections=f"{instrument}/raw/all",
+    )
+
+    logger.info("Handling exposure: %s, %s (%s)", exp.id, exp.obs_id, len(refs))
+
+    source_uri_dir = source_butler.getURI(refs[0]).dirname()
+    logger.debug("Source directory: %s", source_uri_dir)
+
+    # Make a zip file for this exposure
+    with tempfile.TemporaryDirectory() as tmpdir:
+        prepdir = os.path.join(tmpdir, "inputs")
+        os.mkdir(prepdir)
+        os.chdir(prepdir)
+        if dest_path.exists():
+            logger.info("Zip exists, not retrieving datasets: %s", dest_path)
+            return
+
+        # Get the raw datasets
+        start_time = time.time()
+        logger.debug("Retrieving artifacts")
+        retrieved = source_butler.retrieveArtifacts(
+            refs, destination=prepdir, preserve_path=False
+        )
+        logger.debug("retrieved in %s sec", time.time() - start_time)
+
+        # Generate the index
+        _index, okay, failed = index_files(
+            [f.basename() for f in retrieved],
+            "",
+            -1,
+            False,
+            "metadata",
+        )
+        logger.debug("indexed")
+        # ingest-raws needs to be changed to understand this change from
+        # the default of _index.json.
+        with open("_metadata_index.json", "w") as fd:
+            json.dump(_index, fd)
+        logger.debug("index written")
+
+        # Transfer non-dataset files
+        transfer_list = []
+        for dirpath, dirnames, filenames in source_uri_dir.walk():
+            for f in filenames:
+                if not os.path.exists(f):
+                    transfer_list.append((dirpath.join(f), ResourcePath(f)))
+        if dest_path.exists():
+            logger.info("Zip exists, not copying others: %s", dest_path)
+            return
+        logger.debug("Also copying %s", [t[0] for t in transfer_list])
+        ResourcePath.mtransfer("copy", transfer_list)
+
+        # Zip everything.
+        zip_path = os.path.join(tmpdir, zip_name)
+        logger.debug("Writing to %s", zip_path)
+        with zipfile.ZipFile(zip_path, "w") as zip_file:
+            for f in os.listdir():
+                logger.debug("adding %s", f)
+                if f.endswith(".fits"):
+                    zip_file.write(f, f, compress_type=zipfile.ZIP_STORED)
+                else:
+                    zip_file.write(f, f, compress_type=zipfile.ZIP_DEFLATED)
+
+        # Compute the Rucio hashes
+        if config.rucio_rse:
+            hashes = RucioInterface.compute_hashes(zip_path)
+
+        # Copy to destination
+        if dest_path.exists():
+            logger.info("Zip exists, not installing: %s", dest_path)
+            return
+        logger.info("Installing zip in %s", dest_path)
+        if not config.dry_run:
+            dest_path.transfer_from(ResourcePath(zip_path), "copy")
```

Comment:
You seem to be wanting `overwrite=False` here.

Reply:
`overwrite=False` is the default, isn't it?

Reply:
Yes. Sorry.